core/fake/usermanager.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fake
+
+import (
+	"context"
+
+	tpb "github.com/google/keytransparency/core/api/type/type_proto"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// keysetID uniquely identifies a keyset.
+type keysetID struct {
+	instance int64
+	domainID string
+	appID    string
+}
+
+// KeySets implements storage.UserManagerTable in memory.
+type KeySets struct {
+	keysets map[keysetID]*tpb.KeySet
+}
+
+// NewKeySets produces a fake implemenation of storage.UserManagerTable.
+func NewKeySets() *KeySets {
+	return &KeySets{
+		keysets: make(map[keysetID]*tpb.KeySet),
+	}
+}
+
+// Get returns the requestsed keyset.
+func (k *KeySets) Get(ctx context.Context, instance int64, domainID, appID string) (*tpb.KeySet, error) {
+	ks, ok := k.keysets[keysetID{
```

Comment:
The code is a bit stuttery with `keysets[keysetID{}]`

If you could get away with just `ID` that'd be nicer to read; however you're in package `fake` which is a bit broad covering all of `core`, so perhaps `ID` is then too much of a landgrab.

Reply:
how's `kID`?